### ISSUE TITLE
Support cross-account AWS EC2 instance discovery 

### ISSUE THREAD

**Issue by oshati** [gravitational/teleport#23452]
What would you like Teleport to do?

Add support for cross-account AWS ec2 discovery

Currently the Teleport ec2 discovery feature only supports for ec2 resources in a single AWS account but limited when considering auto-discovery of EC2 instances across multiple AWS accounts. 

What problem does this solve?
Reduces overhead with having to manage resources

If a workaround exists, please include it.
Deploy one discovery agent per account

**Comment by code-dave** [comment]
Yes please, this would be a great feature to add. Currently I have to deploy x5 the amount of servers, due to this limitation.

**Comment by philip-teleport** [comment]
+1 from a prospect.

**Comment by nika-chkhikvishvili** [comment]
So 'auto-discovery' needs manual setup for each AWS account? Cross-account IAM has existed since the Jurassic period of AWS. Right now my 'auto' solution requires more manual work than if I just wrote scripts. Please make 'auto' actually mean auto.
